app/src/main/java/org/openobservatory/ooniprobe/activity/add_descriptor/adapter/AddDescriptorExpandableListAdapter.kt

```diff
@@ -0,0 +1,187 @@
+package org.openobservatory.ooniprobe.activity.add_descriptor.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseExpandableListAdapter
+import android.widget.ImageView
+import android.widget.TextView
+import com.google.android.material.checkbox.MaterialCheckBox
+import com.google.android.material.checkbox.MaterialCheckBox.STATE_CHECKED
+import com.google.android.material.checkbox.MaterialCheckBox.STATE_INDETERMINATE
+import com.google.android.material.checkbox.MaterialCheckBox.STATE_UNCHECKED
+import org.openobservatory.engine.OONIRunNettest
+import org.openobservatory.ooniprobe.R
+import org.openobservatory.ooniprobe.activity.add_descriptor.AddDescriptorViewModel
+import org.openobservatory.ooniprobe.test.test.AbstractTest
+
+class GroupedItem(
+    override var name: String,
+    override var inputs: List<String>?,
+    var selected: Boolean = false
+) : OONIRunNettest(name = name, inputs = inputs)
+
+class AddDescriptorExpandableListAdapter(
```

Comment:
Please, add an high level description for this class

Reply:
Sure